### ISSUE TITLE
[bug]: Unified Canvas Invoke ignores the mask when you erase something under the mask on Base layer

### ISSUE THREAD

**Issue by odragora** [invoke-ai/InvokeAI#2157]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### OS

Windows

### GPU

cuda

### VRAM

16

### What happened?

When I mask an area in the bounding box of the Unified Canvas and then erase a part of the image on the Base layer on the area covered by the mask and then Invoke, the mask is ignored.

The new image has changes far outside the masked area. 
Also, the changes are very significant, and they ignore Image To Image Strength value.

**Comment by odragora** [comment]
I also noticed that sometimes it happens even without using eraser or painting on the base layer at all.
Although not as often.
I'm not seeing the pattern yet.

**Comment by odragora** [comment]
I tried to download the image as a file, clear canvas, upload the image I downloaded and set it as the canvas image.
Then I tried to Invoke it multiple times with Image to Image Strength values from 0.01 to 0.99, CFG Scale values from 2 to 20, and different samplers.
Without drawing a mask.

The result was always pretty much the same.
It was different from the original image in the very same details.
It seems like all those values are ignored.

Then I added a mask and invoked again.
The area under the mask has been changed as expected, but the picture outside the mask changed again in the very same way.

I noticed that when I press Invoke, the progress bar with the "Generating (Outpainting)" is being filled twice.
I suspect that there are two different processes of generating are happening because of the bug, and the one that affects the picture outside of the mask should not be happening.
